src/main/scala/Syntax.scala

```diff
@@ -112,12 +112,15 @@ object Syntax {
       case _:EVar | _:EArrAccess => true
       case _ => false
     }
+    var typ: Option[Type] = None
   }
   case class EInt(v: Int, base: Int = 10) extends Expr
   case class EFloat(f: Float) extends Expr
   case class EBool(v: Boolean) extends Expr
   case class EBinop(op: BOp, e1: Expr, e2: Expr) extends Expr
-  case class EArrAccess(id: Id, idxs: List[Expr]) extends Expr
+  case class EArrAccess(id: Id, idxs: List[Expr]) extends Expr {
+    var dims: Option[List[(Int, Int)]] = None
```

Comment:
I don't think you need this. Each `Id` in the program should be decorated with it's type. The `id` here should get a `TArray` type with all the relevant information you want.

Reply:
You're right :+1: